tests/test_smoke.py

```diff
@@ -2297,6 +2318,25 @@ def test_aws_zero_quota_failover():
     run_one_test(test)
 
 
+@pytest.mark.gcp
+def test_gcp_zero_quota_failover():
+
+    name = _get_cluster_name()
+    region = get_gcp_region_for_quota_failover()
+
+    if not region:
+        return
```

Comment:
One thought here - we run the risk of silently passing this test if there's a bug in `check_quota_available` that always returns quota available.

Can we at the least raise a visible warning saying "Unable to test zero quota failover optimization - quotas for {resource} were found on all regions. Is this expected for your account?" if we take this exit? This warning could be a [warnings.warn message](https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html).